zfsonlinux/zfs
# zfs-0.7.7 patchset (squashed)

### Description

This is a squashed patchset for zfs-0.7.7.  The individual commits are in the `tonyhutter:zfs-0.7.7-hutter` branch.  I squashed the commits so that buildbot wouldn't have to run against each one, and because github/buillbot seem to have a maximum limit of 30 commits they can test in a PR.

- Fix MMP write frequency for large pools #7205 #7289
- Handle zio_resume and mmp => off #7286
- Fix zfs-kmod builds when using rpm >= 4.14 #7284
- zdb and inuse tests don't pass with real disks #6939 #7261
- Take user namespaces into account in policy checks #6800 #7270
- Detect long config lock acquisition in mmp #7212
- Linux 4.16 compat: get_disk_and_module() #7264
- Change checksum & IO delay ratelimit values #7252
- Increment zil_itx_needcopy_bytes properly #6988 #7176
- Fix some typos #7237
- Fix zpool(8) list example to match actual format #7244
- Add SMART self-test results to zpool status -c #7178
- Add scrub after resilver zed script #4662 #7086
- Fix free memory calculation on v3.14+ #7170
- Report duration and error in mmp_history entries #7190
- Do not initiate MMP writes while pool is suspended #7182
- Linux 4.16 compat: use correct *_dec_and_test()
- Allow modprobe to fail when called within systemd #7174
- Add SMART attributes for SSD and NVMe #7183 #7193
- Correct count_uberblocks in mmp.kshlib #7191
- Fix config issues: frame size and headers #7169
- Clarify zinject(8) explanation of -e #7172
- OpenZFS 8857 - zio_remove_child() panic due to already destroyed parent zio #7168
- 'zfs receive' fails with "dataset is busy" #7129 #7154
- contrib/initramfs: add missing conf.d/zfs #7158
- mmp should use a fixed tag for spa_config locks #6530 #7155
- Handle zap_add() failures in mixed case mode #7011 #7054
- Fix zdb -ed on objset for exported pool #7099 #6464
- Fix zdb -E segfault #7099
- Fix zdb -R decompression #7099 #4984
- Fix racy assignment of zcb.zcb_haderrors #7099
- Fix zle_decompress out of bound access #7099
- Fix zdb -c traverse stop on damaged objset root #7099
- Linux 4.11 compat: avoid refcount_t name conflict #7148
- Linux 4.16 compat: inode_set_iversion() #7148
- OpenZFS 8966 - Source file zfs_acl.c, function zfs_aclset_common contains a use after end of the lifetime of a local variable #7141
- Remove deprecated zfs_arc_p_aggressive_disable #7135
- Fix default libdir for Debian/Ubuntu #7083 #7101
- Bug fix in qat_compress.c for vmalloc addr check #7125
- Fix systemd_ RPM macros usage on Debian-based distributions #7074 #7100
- Emit an error message before MMP suspends pool #7048
- ZTS: Fix create-o_ashift test case #6924 #6977
- Fix --with-systemd on Debian-based distributions (#6963) #6591 #6963
- Remove vn_rename and vn_remove dependency zfsonlinux/spl#648 #6753
- Fix "--enable-code-coverage" debug build #6674
- Update codecov.yml #6669
- Add support for "--enable-code-coverage" option #6670
- Make "-fno-inline" compile option more accessible #6605
- Add configure option to enable gcov analysis #6642
- Implement --enable-debuginfo to force debuginfo #2734
- Make --enable-debug fail when given bogus args #2734

### Motivation and Context



### How Has This Been Tested?





### Types of changes


### Checklist:


- [ ] My code follows the ZFS on Linux code style requirements.
- [ ] All commit messages are properly formatted and contain `Signed-off-by`.
- [ ] Change has been approved by a ZFS on Linux member.
